main.cpp

```diff
@@ -19,15 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult &Result) override 
+    {
+        const auto *expression = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        
+        auto range_replace = CharSourceRange::getCharRange(expression->getLParenLoc(),
+                             expression->getSubExprAsWritten()->getBeginLoc());
+        auto &src_manage = *Result.SourceManager;
+
+        auto type_name = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                                              expression->getLParenLoc().getLocWithOffset(1),
+                                              expression->getRParenLoc().getLocWithOffset(-1)),
+                                              src_manage, LangOptions());
+        const auto *expr = expression->getSubExprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?
Please provide examples.

Reply:
getSubExpr() retrieves the first cast subexpression
getSubExprAsWritten() retrieves the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis.
ignoreImpCasts() skips past any implicit casts which might surround this expression until reaching a fixed point. It skips the following AST nodes: ImplicitCastExpr, FullExpr.

llvm::outs() << "getSubExpr()->ignoreImpCasts(): " << castNode->getSubExpr()->ignoreImpCasts()->getStmtClassName();
llvm::outs() << "getSubExprAsWritten(): " << castNode->getSubExprAsWritten()->getStmtClassName();
llvm::outs() << "getSubExprAsWritten()->ignoreImpCasts(): " << castNode->getSubExprAsWritten()->IgnoreImpCasts()->getStmtClassName();

Output will be:
getSubExpr()->ignoreImpCasts(): DeclRefExpr
getSubExprAsWritten()->IgnoreImpCasts(): DeclRefExpr

getSubExprAsWritten(): DeclRefExpr

Reply:
You didn't describe the reason why you are using `getSubExprAsWritten()` and `IgnoreImpCasts()`.

Reply:
> You didn't describe the reason why you are using `getSubExprAsWritten()` and `IgnoreImpCasts()`.

I used IgnoreImpCasts() just to be sure there are no implicit casts left, but as the example shows, it wasn't necessary.